gumps/VideoOptions_gump.cc

```diff
@@ -206,7 +207,7 @@ void VideoOptions_gump::rebuild_dynamic_buttons() {
 		// the text arrays is freed by the destructor of the button
 		std::vector<std::string> scalingtext;
 		for (int i = 0; i < num_scales; i++) {
-			char buf[10];
+			char buf[16];
 			snprintf(buf, sizeof(buf), "x%d", i + 1);
 			scalingtext.emplace_back(buf);
```

Comment:
I would recommend to do away with `buf` altogether and replace the entire loop body by this:
```c++
scalingtext.push_back('x' + std::to_string(i + 1));
```